telemetry/statsd/src/main/scala/io/buoyant/telemetry/statsd/StatsDStatsReceiver.scala

```diff
@@ -0,0 +1,44 @@
+package io.buoyant.telemetry.statsd
+
+import java.util.concurrent.ConcurrentHashMap
+import com.timgroup.statsd.StatsDClient
+import com.twitter.finagle.stats.{Counter, Stat, StatsReceiverWithCumulativeGauges}
+import scala.collection.JavaConverters._
+
+private[telemetry] object StatsDStatsReceiver {
+  // from https://github.com/researchgate/diamond-linkerd-collector/
+  private[statsd] def mkName(name: Seq[String]): String = {
+    name.mkString("/")
+      .replace(".", "_")
+      .replace("#", "")
+      .replace("//", "/")
+      .replace("/", ".") // http://graphite.readthedocs.io/en/latest/feeding-carbon.html#step-1-plan-a-naming-hierarchy
+  }
+}
+
+private[telemetry] class StatsDStatsReceiver(statsDClient: StatsDClient)
+  extends StatsReceiverWithCumulativeGauges {
+  import StatsDStatsReceiver._
+
+  val repr: AnyRef = this
+
+  private[statsd] def flushGauges(): Unit = gauges.values.foreach(_.send)
+  private[statsd] def close(): Unit = statsDClient.stop()
+
+  private[this] val gauges = new ConcurrentHashMap[String, Metric.Gauge].asScala
```

Comment:
From 
https://github.com/scala/scala/blob/v2.9.2/src/library/scala/collection/JavaConverters.scala#L330
`The returned Scala <code>Iterator</code> is backed by the provided Java <code>Iterator</code>`

Also, the `ConcurrentHashMap` section of https://twitter.github.io/scala_school/concurrency.html seems to imply thread-safety.